crates/swc_error_reporters/src/handler.rs

```diff
@@ -114,6 +114,15 @@ where
             Ok(_) => Err(anyhow::anyhow!(msg)),
             Err(err) => Err(err.context(msg)),
         }
+    } else if handler.has_warning() {
+        let mut lock = wr.0.lock();
+        let error = take(&mut *lock);
+
+        let msg = String::from_utf8(error).expect("error string should be utf8");
+
+        println!("{}", msg);
```

Comment:
We should not print it. Assuming that the user is going to print is too much of an assumption.
I think one solution is to return the warning string in the `TransformOutput` instead, although it may require a breaking change.

Reply:
Yes, your worry is considerate.
Emit warning string in the `TransformOutput` instead is the best way to resolve it. But we need more discuss it that avoid swc import uncomfortable breaking change. 
I will try another way to fix it.

Reply:
How about defining a trait in `swc_error_reporters` and implementing it for `TransformOutput`?

Reply:
We need create a new fields in `TransformOutput` to save diagnostics, then node cli print it using js api.

Reply:
Yeah that's what I intended. You can't avoid a breaking change in this case, but you can define a trait in `swc_error_reporters` to mitigate the dependency issue